src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -424,14 +461,16 @@ private boolean reauthorize(
 
                 if (reauthorized)
                 {
-                    expiresAt = expiresAtMillis;
+                    this.expiresAt = expiresAtMillis;
+                    this.challengeResponseTimeout = challengeResponseTimeout;
                 }
             }
             else
             {
                 this.subject = subject != null ? subject.intern() : null;
                 this.authorization = connectAuthorization;
                 this.expiresAt = expiresAtMillis;
+                this.challengeResponseTimeout = challengeResponseTimeout;
                 reauthorized = false;
```

Comment:
Initialize `reauthorized` to `false` and non-final, then only overwrite if attempted when `referenceCount > 0`.